pil-analyzer/src/pil_analyzer.rs

```diff
@@ -257,6 +258,69 @@ impl PILAnalyzer {
         }
     }
 
+    pub fn structs_check(&self) -> Result<(), Vec<Error>> {
+        let mut errors = vec![];
+        let struct_declarations: HashMap<String, StructDeclaration> = self
+            .definitions
+            .iter()
+            .filter_map(|(name, (_, value))| {
+                if let Some(FunctionValueDefinition::TypeDeclaration(TypeDeclaration::Struct(
+                    struct_decl,
+                ))) = value
+                {
+                    Some((name.clone(), struct_decl.clone()))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        for (_, def) in self.definitions.values() {
```

Comment:
This is not enough, we also need to traverse proof items and impls. Maybe the easiest is to implement `Children<Expression>` on `PilAnalyzer` and then call `self.all_children()`.